iroh-sync/src/ranger.rs

```diff
@@ -400,13 +400,17 @@ where
 
     /// Processes an incoming message and produces a response.
     /// If terminated, returns `None`
+    ///
+    /// `validate_cb` is called before an entry received from the remote is inserted into the store.
+    /// It must return true if the entry is valid and should be stored, and false otherwise
+    /// (which means the entry will be dropped and not stored).
     pub fn process_message<F>(
         &mut self,
         message: Message<K, V>,
-        cb: F,
+        validate_cb: F,
     ) -> Result<Option<Message<K, V>>, S::Error>
     where
-        F: Fn(K, V),
+        F: Fn(&S, K, V) -> bool,
```

Comment:
Could we pass references for all here? That way we don't always have to clone them